lib/_http_agent.js

```diff
@@ -319,6 +330,20 @@ function installListeners(agent, s, options) {
   }
   s.on('close', onClose);
 
+  function onTimeout() {
+    debug('CLIENT socket onTimeout');
+
+    // Destroy if in free list.
+    // TODO(ronag): Always destroy, even if not in free list.
+    const sockets = agent.freeSockets;
+    for (const name of ObjectKeys(sockets)) {
+      if (sockets[name].includes(s)) {
+        return s.destroy();
```

Comment:
should remove the destroy socket from freeSockets list immediately to prevent new requests from being sent through this socket. 

```js
if (sockets[name].includes(s)) {
  s.destroy();
  return agent.removeSocket(s, options);
}
```